libkpod/image/copy.go

```diff
@@ -100,8 +104,115 @@ func PushImage(srcName, destName string, options CopyOptions) error {
 	return nil
 }
 
+type imagePullStruct struct {
+	imageName   string
+	tag         string
+	registry    string
+	hasRegistry bool
+	transport   string
+}
+
+func (ips imagePullStruct) returnFQName() string {
+	fqname := fmt.Sprintf("%s%s/%s:%s", ips.transport, ips.registry, ips.imageName, ips.tag)
+	return fqname
+}
+
+func imageBeginsWithTransport(image string) bool {
+	parts := strings.SplitN(image, ":", 2)
+	if len(parts) == 1 {
+		return false
+	}
+	for _, transport := range transports.ListNames() {
+		if strings.HasPrefix(parts[0], transport) {
+			return true
+		}
+	}
+	return false
+}
+
+func getRegistriesToTry(image string) ([]types.ImageReference, error) {
+	var registries []types.ImageReference
+	var imageError = fmt.Sprintf("Unable to parse '%s'\n", image)
+	imgRef, err := reference.Parse(image)
+	if err != nil {
+		return registries, errors.Errorf(imageError)
+
```

Comment:
nit, I'd flip line 139/140 so you'd the blank line after the bracket

Reply:
yup